node.go

```diff
@@ -122,12 +125,16 @@ func (node *Node) HealthCheck() (bool, error) {
 		return false, err
 	}
 	node.getNodeResponse(nil, err)
+	node.lock.RLock()
+	defer node.lock.RUnlock()
 	return node.IsHealthy, nil
```

Comment:
I don't know the answer to this question - but I suspect the if all you are protecting here is single value read/writes (of `isHealthy` and `LastHealthCheck` that you could use atomic read writes from `atomic` and that that might be more efficient than acquiring a lock and releasing it.